cdap-ui/app/cdap/components/DataPrep/Directives/MapToTarget/index.js

```diff
@@ -14,36 +14,367 @@
 
-  render() {
+  const highlightText = (text) => {
+    const searchTextUpper = searchText.trim().toUpperCase();
+    if (!searchTextUpper) {
+      return text;
+    }
+    const index = text.toUpperCase().indexOf(searchTextUpper);
+    if (index < 0) {
+      return text;
+    }
+    const leadingText = text.substring(0, index);
+    const highlightedText = text.substring(index, index + searchTextUpper.length);
+    const trailingText = text.substring(index + searchTextUpper.length);
     return (
-      <div
-        id='map-to-target-directive'
-        className={classnames('clearfix action-item', {
-          active: this.props.isOpen && !this.props.isDisabled,
-          disabled: this.props.isDisabled,
-        })}
-      >
-        <span>{T.translate(`${PREFIX}.title`)}</span>
-        <span className='float-right'>
-          <span className='fa fa-caret-right' />
-        </span>
+      <span>
+        {leadingText}
+        <span className={classes.highlight}>{highlightedText}</span>
+        {trailingText}
+      </span>
+    );
+  };
+
+  const resetTargetOptionsScroll = () => {
+    const element = document.querySelector('.' + classes.targetOptionList);
+    if (element) {
+      element.scrollTop = 0;
+    }
+  };
+
+  const selectTargetDataModel = async (dataModel) => {
+    setLoading(`${PREFIX}.loadingText`);
+    try {
+      setTargetModel(null);
+      await setTargetDataModel(dataModel);
+      setSearchText('');
+      resetTargetOptionsScroll();
+    } catch (error) {
+      setError(error, 'Could not set target data model');
+    } finally {
+      setLoading('');
+    }
+  };
+
+  const selectTargetModel = (model) => {
+    setTargetModel(model);
+    setSearchText('');
+    resetTargetOptionsScroll();
+  };
+
+  const applyDirective = async (field) => {
+    setLoading(`${PREFIX}.executingDirectiveText`);
+    try {
+      await saveTargetDataModelStates();
+
+      const directive = 'data-model-map-column ' +
+        `'${targetDataModel.url}' '${targetDataModel.id}' ${targetDataModel.revision} ` +
+        `'${targetModel.id}' '${field.id}' :${column}`;
+
+      await execute([directive], false, true).toPromise();
+
+      close();
+      onComplete();
+    } catch (error) {
+      setError(error, 'Error executing Map to Target directive');
+    } finally {
+      setLoading('');
+    }
+  };
+
+  const renderHeader = (selection) => {
+    return (
+      <div>
+        {selection.length === 0 ? <h5>{T.translate(`${PREFIX}.dataModelPlaceholder`)}</h5> : null}
+        {selection.map(item => (
+          <div id={`map-to-target-selected-${item.key}`} key={item.key} className={classes.selectedItem}>
+            <span className={classes.selectedItemLabel}>{item.label}:&nbsp;</span>
+            <span className={classes.selectedItemName}>{item.name}</span>
+            {!loading // Hide [X] when loading
+              ? <span className={classnames('fa fa-times', classes.unselectIcon)} onClick={item.unselectFn} />
```

Comment:
Here as well. Please use `<IconSVG />` component to render icons.

Reply:
Done.